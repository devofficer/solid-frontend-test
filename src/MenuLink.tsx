type Props = {
  text: string,
  icon: string,
  color?: string,
}

const MenuLink = ({ icon, text, color = '' }: Props) => (
  <div className={`nav__dropdown-menu-item ${color}`}>
    <i className="material-icons-outlined">
      {icon}
    </i>

    {text}
  </div>
)

export default MenuLink
