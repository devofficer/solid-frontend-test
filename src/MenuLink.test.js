import { shallow } from 'enzyme'
import MenuLink from './MenuLink'

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
}))

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
}))

describe('<MenuLink />', () => {
  it('renders link', () => {
    expect(shallow(
      <MenuLink icon="phone" text="Call me" path="/" />
    )).toMatchSnapshot()
  })
})
