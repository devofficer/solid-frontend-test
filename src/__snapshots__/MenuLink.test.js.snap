// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> renders link 1`] = `
<div
  className="nav__dropdown-menu-item"
>
  <i
    className="material-icons-outlined"
  >
    phone
  </i>
  Call me
</div>
`;
